9956: do not test `contains` result against S.true

smichr opened the pull request:
The test added identifies and issue with testing with "is true" instead of "== True" in the sets module. 
In master, the Intersection gives EmptySet because geometry returns True rather than true.

I have tried to change all such cases in sets to use True/False instead of assume that the `true` value will be returned by any comparison.

smichr committed:
9956: do not test contains against S.true

smichr commented:
ping @gxyd

gxyd commented on sympy/sets/sets.py:
```diff
@@ -1002,7 +1002,7 @@ def _eval_imageset(self, f):
             domain_set = self
             for (p_expr, p_cond) in expr.args:
-                if p_cond is S.true:
+                if p_cond is true:
```
Any reason for not using `==` here?

smichr replied:
Piecewise sympifies its args so any True is converted to S.true:

```
>>> Piecewise((x,x>0),(1,True))
Piecewise((x, x > 0), (1, True))
>>> _.args
((x, x > 0), (1, True))
>>> type(_[1][1])
<class 'sympy.logic.boolalg.BooleanTrue'>
```

gxyd commented on sympy/sets/tests/test_sets.py:
```diff
@@ -453,6 +453,12 @@ def test_contains():
     assert not RootOf(x**5 + x**3 + 1, 1) in S.Reals
 
+    # non-bool results
+    assert Union(Interval(1, 2), Interval(3, 4)).contains(x) == \
```
These tests work the same in master. Do these test something different?

smichr replied:
No, they just establish that no change should lose these elements by, perhaps, just testing for true and false values: one must account for the Basic case.

gxyd commented on sympy/sets/tests/test_sets.py:
```diff
@@ -453,6 +453,12 @@ def test_contains():
+    # non-bool results
+    assert Union(Interval(1, 2), Interval(3, 4)).contains(x) == \
+        Or(And(x <= 2, x >= 1), And(x <= 4, x >= 3))
+    assert Intersection(Interval(1,x),Interval(2,3)).contains(y) == \
```
Perhaps space `1,x` -> `1, x` and `2,3` -> `2, 3` could be added

smichr replied:
done -- sorry I missed that.

smichr committed:
changed slow decoration of tests

I don't know why refine is being used on the test in 8545. There is
nothing in the issue about it and it takes a long time to do nothing.

>>> var('x')
x
>>> eq = 1 - x - sqrt((1 - x)**2);eq
-x - sqrt((-x + 1)**2) + 1
>>> refine(eq)
-x - sqrt((-x + 1)**2) + 1

gxyd commented:
You have merges conflicts here. The PR looks good to me.

smichr committed:
Merge branch 'master' of github.com:sympy/sympy
